lte/gateway/python/magma/pipelined/app/enforcement_stats.py

```diff
@@ -574,29 +583,31 @@ def _get_rule_id(self, flow):
 
     def get_stats(self, cookie: int = 0, cookie_mask: int = 0):
         """
-        Use Ryu API to send a stats request containing cookie and cookie mask, retrieve a response and 
+        Send a stats request containing cookie and cookie mask,
+        wait for response from OVS using a future, retrieve a response and
         convert to a Rule Record Table and remove old flows
         """
         if not self._datapath:
             self.logger.error("Could not initialize datapath for stats retrieval")
             return RuleRecordTable()
-        parser = self._datapath.ofproto_parser
-        message = parser.OFPFlowStatsRequest(datapath=self._datapath, cookie = cookie, cookie_mask = cookie_mask)
         try:
-            response = ofctl_api.send_msg(self, message, reply_cls=parser.OFPFlowStatsReply,
-                    reply_multi=False)
-            if response == None:
+            xid = flows.send_stats_request(self._datapath, self.tbl_num, cookie,
+                                           cookie_mask)
+            self._poll_futures[xid] = Future()
+            res = self._poll_futures[xid].result(timeout=self._stats_wait_timeout)
```

Comment:
Can you delete the map entry once you are done with it to avoid holding ref to the object already processed.

Reply:
right, I had that in mind but forgot to do it. Thanks

Reply:
added